packages/astro/src/core/cookies/cookies.ts

```diff
@@ -0,0 +1,236 @@
+import type { CookieSerializeOptions } from 'cookie';
+import { parse, serialize } from 'cookie';
+import ms from 'ms';
+
+interface AstroCookieSetOptions {
+  domain?: string;
+  expires?: number | Date | string;
+  httpOnly?: boolean;
+  maxAge?: number;
+  path?: string;
+  sameSite?: boolean | 'lax' | 'none' | 'strict';
+  secure?: boolean;
+}
+
+interface AstroCookieDeleteOptions {
+	path?: string;
+}
+
+interface AstroCookieInterface {
+  value: string | undefined;
+  json(): Record<string, any>;
+  number(): number;
+	boolean(): boolean;
+}
+
+interface AstroCookiesInterface {
+  get(key: string): AstroCookieInterface;
+	has(key: string): boolean;
+  set(key: string, value: string | Record<string, any>, options?: AstroCookieSetOptions): void;
+  delete(key: string, options?: AstroCookieDeleteOptions): void;
+}
+
+const DELETED_EXPIRATION = new Date(0);
+const DELETED_VALUE = 'deleted';
+
+class AstroCookie implements AstroCookieInterface {
+	constructor(public value: string | undefined) {}
+	json() {
+		if(this.value === undefined) {
+			throw new Error(`Cannot convert undefined to an object.`);
+		}
+		return JSON.parse(this.value);
```

Comment:
Could this error? If I’ve followed the logic:

1. User: `cookies.set('foo', 'bar')`
   - `AstroCookies` checks the value is a string, and stores `'bar'` as the serialized value

2. User: `cookies.get('foo').json()`
   - Results in `JSON.parse('bar')`, which throws `Uncaught SyntaxError`

Reply:
If we know we’re always running `JSON.stringify` on non-strings, probably safe to return the raw value if parse fails?

```suggestion
		try {
			return JSON.parse(this.value);
		} catch {
			return this.value;
		}
```

Reply:
(Also possible that this error-ing is intentional — don’t call `.json()` if you haven’t stored an object.)